### PR TITLE
Further coverage for compression

### PR DESCRIPTION
**Pull request by dgquintas**
Hitting https://github.com/grpc/grpc/compare/master...dgquintas:compression_coverage_2?expand=1#diff-7fafa5ac05a9939cf1c9b30b01c8e828R73 remains ellusive though...

### CODE REVIEW

**Comment by ctiller** [comment]
Can you take a few kb from /dev/random and try to decompress it as gzip?

**Comment by nicolasnoble** [comment]
Not really portable :-)

**Comment by ctiller** [comment]
Copying it into a string and embedding it in the test is.

**Comment by ctiller** [comment]
We need only one bad test vector.

**Comment by ctiller** [comment]
I left some Python code in tools/codegen/core today to do the escaping...
Marry that with the read from random from some other Python script.

**Comment by dgquintas** [comment]
No dice. Trying with random data (different each time, using rand()) still goes out the "zlib: not all input consumed" branch

File: test/core/compression/compression_test.c [hunk 1]
```diff
@@ -71,9 +70,93 @@ static void test_compression_algorithm_parse(void) {
...
+    GRPC_COMPRESS_DEFLATE, GRPC_COMPRESS_DEFLATE, GRPC_COMPRESS_DEFLATE};
+  gpr_log(GPR_DEBUG, "test_compression_algorithm_for_level");
+
+  for (i = 0; i < GPR_ARRAY_SIZE(levels); i++) {
+    GPR_ASSERT(algorithms[i] ==
+               grpc_compression_algorithm_for_level(levels[i]));
+  }
+}
+
+static void test_compression_level_for_algorithm(void) {
```
Comment (ctiller) [inline test/core/compression/compression_test.c]: Lets kill this function (grpc_compression_level_for_algorithm) - it's impossible to use correctly

**Comment by ctiller** [comment]
LGTM. Merge at will.

**Pull request merged**